src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -615,7 +615,7 @@ object IndexLogEntry {
 /**
  * Provides functionality to generate unique file ids for files.
  */
-class FileIdTracker {
+private[hyperspace] class FileIdTracker {
```

Comment:
Then I think we should change the signature of getFileToIdMap to return Seq, because with the current one users can mess up the map

Reply:
Removed private and changed the return type of getFileToIdMap to return Seq, not Map. This is to protect the class from misuse - please see the commit comment.

There is another option: instead of making lineagePairs take FileIdTracker as an argument, we can request a file name mapper from the implementation, because every usage of the tracker is in this form:

```scala
fileIdTracker.getFileToIdMap.map { kv => (<some replace function>(kv._1._1), kv._2) }
```

By the way, I feel a little bit confused what parts of the code is considered public and extendable for users. Shouldn't we keep the public API and extension points clean and minimal? Currently many classes are not marked private and they can be misused by users.

Reply:
It's for extensible source provider - so that a user can implement & add a new data source like Delta Lake or Iceberg, without modifying Hyperspace code. Currently, we don't expose APIs returning IndexLogEntry directly (in Hyperspace.scala), so fileIdTracker would be ok unless they use a custom build.

Reply:
> There is another option: instead of making lineagePairs take FileIdTracker as an argument, we can request a file name mapper from the implementation

I like this idea. Do you want to try this option?

Reply:
@imback82 Yes, but maybe in another PR as it will change the API which is not related to FileIdTracker significantly?

Reply:
yes, definitely as a separate PR (sorry, I wan't clear 😄 )